mrp_unbuild_tracked_raw_material/models/unbuild.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2019 Akretion (http://www.akretion.com). All Rights Reserved
+# @author David BEAL <david.beal@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as DT_FORMAT
+
+logger = logging.getLogger(__name__)
+
+MESSAGE = "Some of your components are tracked, you have to specify " \
+          "a manufacturing order in order to retrieve " \
+          "the correct components."
+ALTER_MESSAGE = "Alternatively, you may unbuild '%s' tracked product " \
+                "if you set it as '%s'.\n" \
+                "In this case lots'll be automatically created for you."
+
+
+class MrpUnbuild(models.Model):
+    _inherit = "mrp.unbuild"
+
+    def action_unbuild(self):
+        """ We need to catch raise behavior when tracked products
+            are unbuild without an original manufacturing order and
+            go on with another workflow with _action_unbuild_alternative_way()
+        """
+        try:
+            res = super().action_unbuild()
+        except UserError as err:
+            # Unbuild is impossible because of MESSAGE
+            if hasattr(err, 'name') and err.name == _(MESSAGE):
```

Comment:
I think it's cleaner to check the conditions instead of the error message.

Reply:
There are several UserError in the original method. I'm not really sure it could be more reliable with conditions because they could change in the original module (I can control that).